test/ParserTest.php

```diff
@@ -119,4 +119,22 @@ public function testCancelReturnsInternalBuffer() {
 
         $this->assertSame("d", $parser->cancel());
     }
+
+    public function testIsValidOnValidParser() {
+        $parser = new Parser((function () {
+            yield 3;
+        })());
+
+        $this->assertTrue($parser->isValid());
+    }
+
+    public function testIsValidShouldReturnFalse() {
```

Comment:
Could you change this test name, too, then?

Reply:
Do you have any idea about the new test name?

Reply:
How about ```testIsValidOnValidParserWithPendingValue```?

Reply:
Maybe `testIsValidOnNonFinishedParser()` and `testIsValidOnFinishedParser()`

Reply:
Ok, I think we use ```testIsValidOnNonFinishedParser``` is good.